lib/config/defaults.js

```diff
@@ -581,19 +581,19 @@ const applyOutputDefaults = (
 	});
 	F(output, "chunkFormat", () => {
 		if (tp) {
+			if (tp.document) return "array-push";
```

Comment:
What's the reason behind this change?

Reply:
> and also fixes `electron-renderer` to use the correct chunk format, as it is in a browser and not in a module context.

It's should be in both contexts. So `require()` should be available and we can use it to load chunk.

Reply:
Note that `electron-renderer` means a `BrowserWindow` with `nodeIntegration: true`.

For a `BrowserWindow` with `nodeIntegration: false` you should use `target: "web"`.

Reply:
The primary reason behind this change was to match previous behaviour. As you can see [here](https://github.com/webpack/webpack/commit/7e85649492a996ababa31eb73933e1d3edd476ee#diff-c147c0dd597c04e90f7efaa203b5d0c7L584), `electron-renderer` used to use `array-push` and `jsonp`, and my changes restore that behaviour.

`require()` is available, yes; but `chunkFormat=commonjs` makes it generate a chunk in the format
```js
exports.id = "common";
exports.ids = ["common"];
exports.modules = {
...
```
which doesn't work because there is no module context in the browser and so we get a 
```
common.js:1 Uncaught ReferenceError: exports is not defined
```
regardless of whether `nodeIntegration` is turned on (unless I've overlooked something else, of course).